tasks/service/async/src/main/java/org/projectnessie/nessie/tasks/async/TasksAsync.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.nessie.tasks.async;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Supplier;
+
+public interface TasksAsync {
+  Clock clock();
+
+  default CompletionStage<Void> call(Runnable runnable) {
+    return supply(
+        () -> {
+          runnable.run();
+          return null;
+        });
+  }
+
+  <R> CompletionStage<R> supply(Supplier<R> runnable);
+
+  CompletionStage<Void> schedule(Runnable runnable, Instant scheduleNotBefore);
```

Comment:
See my other comment about cancellation.

Reply:
If we want to call `.cancel()` on the returned object, let's return a `CompletableFuture` to avoid ambiguity.